iina/PlaylistViewController.swift

```diff
@@ -143,6 +153,151 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource {
+      let indexSet = dataArray[0]
+
+      let playlistCount = tableView.numberOfRows - 1
+      var order: Array<Int> = Array(0...playlistCount)
+      var finalRow = row
+
+      for selectedRow in indexSet.reversed() {
+        if selectedRow < row {
+          finalRow -= 1
+        }
+        order.remove(at: selectedRow)
+      }
+
+      for selectedRow in indexSet.reversed() {
+        order.insert(selectedRow, at: finalRow)
+      }
+
+      var fileList: [String] = []
+      var playing: Int = 0
+      for playlistItem in playerCore.info.playlist {
+        fileList.append(playlistItem.filename)
+        if playlistItem.isPlaying {
+          playing = fileList.count - 1
+        }
+      }
+
+      for i in Array(0...playlistCount).reversed() {
+        if i == playing {
+          continue
+        }
+        playerCore.removeFromPlaylist(index: i)
+      }
+
+      var insertPosition = 0
+      var foundPlaying: Bool = false
+
+      for i in 0...playlistCount {
+        if order[i] == playing {
+          foundPlaying = true
+          continue
+        }
+        playerCore.addToPlaylist(fileList[order[i]])
+        if !foundPlaying {
+          playerCore.mpvController.command(.playlistMove, args: [(insertPosition + 1).toStr(), insertPosition.toStr()])
+          insertPosition += 1
+        }
+      }
+
+      NotificationCenter.default.post(Notification(name: Constants.Noti.playlistChanged))
+
+      var finalIndexSet: IndexSet = []
+      for i in 0...playlistCount {
+        if tableView.selectedRowIndexes.contains(order[i]) {
+          finalIndexSet.insert(i)
+        }
+      }
+      tableView.deselectAll(self)
+      tableView.selectRowIndexes(finalIndexSet, byExtendingSelection: false)
+
+      return true
+
+    } else if let fileNames = pasteboard.propertyList(forType: NSFilenamesPboardType) as? [String] {
+      var added = 0
+      var currentRow = row
+      var playlistItems = tableView.numberOfRows - 1
+      fileNames.forEach({ (path) in
+        let ext = (path as NSString).pathExtension
+        if !playerCore.supportedSubtitleFormat.contains(ext) {
+          playerCore.addToPlaylist(path)
+          playlistItems += 1
+          playerCore.mpvController.command(.playlistMove, args: [playlistItems.toStr(), currentRow.toStr()])
+
+          currentRow += 1
+          added += 1
+        }
+      })
+
+      if added == 0 {
+        return false
+      }
+      
+      var finalIndexSet: IndexSet = []
+      let selectedIndexSet: IndexSet = tableView.selectedRowIndexes
+      for i in selectedIndexSet {
+        if i >= row {
+          finalIndexSet.insert(i + added)
+        } else {
+          finalIndexSet.insert(i)
+        }
+      }
+      tableView.deselectAll(self)
+      tableView.selectRowIndexes(finalIndexSet, byExtendingSelection: false)
+
+      NotificationCenter.default.post(Notification(name: Constants.Noti.playlistChanged))
```

Comment:
Maybe no need for this notification, since it's always posted when `playlist/count` changed